packages/plugin-http-client/src/HttpClient.ts

```diff
@@ -0,0 +1,119 @@
+import type {
+  Dependencies,
+  HttpAgent,
+  HttpAgentRequestOptions,
+  UnknownParameters,
+} from '@ima/core';
+
+declare module '@ima/core' {
+  interface OCAliasMap {
+    $DefaultProcessors: Processor[];
+  }
+}
+
+import { Processor, Operation, ProcessorParams } from './Processor';
+
+export const OPTION_TRANSFORM_PROCESSORS = 'transformProcessors';
+
+export type HttpClientRequestOptions = {
+  [OPTION_TRANSFORM_PROCESSORS]?: (processor: Processor[]) => Processor[];
+} & HttpAgentRequestOptions;
+
+export type HttpClientRequestMethod =
+  | 'get'
+  | 'put'
+  | 'post'
+  | 'patch'
+  | 'delete';
+
+export type HttpClientRequest = {
+  method: HttpClientRequestMethod;
+  url: string;
+  data?: UnknownParameters;
+  options?: HttpClientRequestOptions;
+};
+
+export class HttpClient {
+  #http: HttpAgent;
+  #defaultProcessors: Processor[];
+
+  static get $dependencies(): Dependencies {
+    return ['$Http', '...?$DefaultProcessors'];
+  }
+
+  constructor(http: HttpAgent, ...defaultProcessors: Processor[]) {
+    this.#http = http;
+
+    this.#defaultProcessors = defaultProcessors.filter(
+      item => item instanceof Processor
+    );
+  }
+
+  registerProcessor(processor: Processor) {
+    this.#defaultProcessors.push(processor);
+  }
+
+  async request(
```

Comment:
Can we add `request<B = any>` generics, which gets passed to `_runProcessors` ... this should theoretically result in typed body when this method resolves?

Reply:
Added.